feat: add GitHub Action for rollup upgrade

ludamad opened the pull request:
Adds a GitHub Action workflow to deploy a new Rollup version and create a `RegisterNewRollupVersionPayload` for governance approval.

## Usage

Trigger the workflow from GitHub Actions with:
- **network**: `mainnet`, `testnet`, or `devnet`
- **registry_address**: The existing registry contract address

The workflow will:
1. Build the codebase to generate genesis values (VK tree root, protocol contracts hash, etc.)
2. Deploy a new Rollup contract with the current code
3. Create a `RegisterNewRollupVersionPayload` that governance can execute to register the upgrade

## Network Configuration

All L1 contract parameters are defined in `spartan/environments/network-defaults.yml`. When you select a network:

- **Contract constants** (`AZTEC_EPOCH_DURATION`, `AZTEC_SLOT_DURATION`, thresholds, etc.) are loaded from the network's configuration
- **L1 chain** is inferred from `L1_CHAIN_ID` in the config (mainnet uses Ethereum mainnet, testnet/devnet use Sepolia)
- **Credentials** are fetched from GCP secrets based on the L1 chain (`sepolia-rpc-urls`, `sepolia-labs-rollup-private-key`, etc.)

This means the deployment automatically uses the correct parameters for each network without manual configuration.

## Follow-on tasks
- [TMNT-495](https://linear.app/aztec-labs/issue/TMNT-495/add-mainnet-labs-rollup-private-key-gcp-secret-for-rollup-upgrades): Add mainnet-labs-rollup-private-key GCP secret for rollup upgrades

just-mitch commented on .github/workflows/create-rollup-upgrade-payload.yml:
```diff
@@ -0,0 +1,54 @@
+          REGISTRY_ADDRESS: ${{ inputs.registry_address }}
+          ROLLUP_ADDRESS: ${{ inputs.rollup_address }}
+        run: |
+          forge script script/deploy/CreateRollupUpgradePayload.s.sol:CreateRollupUpgradePayload \
```
this needs to `--verify --verifier etherscan --etherscan-api-key XXX`.

ludamad replied:
ah right I missed this as it was a simple payload

just-mitch commented on l1-contracts/script/deploy/CreateRollupUpgradePayload.s.sol:
```diff
@@ -0,0 +1,34 @@
+///   - ROLLUP_ADDRESS: Address of the new Rollup to register
+///
+/// Outputs the deployed payload address.
+contract CreateRollupUpgradePayload is Script {
```
I'm wondering if we should just deploy the new rollup in here. Thoughts @LHerskind ?

just-mitch replied:
The best flow I see is running one github action, and having the rollup and upgrade payload we want deployed.

ludamad replied:
Ah I was wonder, wasn't sure if people would want to first smoke test a deployed rollup then do this

just-mitch replied:
Makes sense, though there is nothing stopping us from doing both and still smoke testing.

just-mitch replied:
I would include the rollup deployment. It will also force us now to define the configuration parameters we want to use.

ludamad commented on l1-contracts/bootstrap.sh:
```diff
@@ -96,7 +96,7 @@ function build_verifier {
       $(find generated -name '*.sol') \
       test/shouting.t.sol \
       script/deploy/*.s.sol \
-      tests/scripts/*.t.sol
```
typo fix

just-mitch commented on .github/workflows/create-rollup-upgrade-payload.yml:
```diff
@@ -0,0 +1,75 @@
+          ./ci.sh deploy-rollup-upgrade \
+            "${{ inputs.rpc_url }}" \
+            "${{ inputs.registry_address }}" \
+            "${{ inputs.private_key }}" \
```
Just sanity checking- this is properly redacted etc?

ludamad replied:
No spot on, not sure what I was thinking. It's best these go through secrets, even if we're changing them for diff networks

just-mitch commented on spartan/scripts/deploy_rollup_upgrade.sh:
Is it possible to call this like `/deploy_rollup_upgrade.sh <rpc_url> <registry_address> <private_key> NETWORK=mainnet`?

ludamad replied:
Yes

ludamad replied:
you will get the rollup contract in the repo and the rollup config values from spartan/environments/network-defaults.yml .l1-contracts section

just-mitch replied:
They wouldn't be overwritten by the values in spartan/environments/network-defaults.yml .networks.mainnet ?

ludamad replied:
Good point. I moved to specifying network and it makes sense to use those

ludamad commented:
@just-mitch did a round of cleanup. I think this is the final form. it should take the network label,, relies on network-defaults.yml having the entries, GCP secrets. So it should just be point to a release, target a network+registry and go

just-mitch reviewed (approved):
Lovely. Thanks for the persistence here @ludamad 👍

ludamad committed:
feat: add GitHub Action for rollup upgrade

Adds a GitHub Action workflow to deploy a new Rollup version and create a `RegisterNewRollupVersionPayload` for governance approval.

## Usage

Trigger the workflow from GitHub Actions with:
- **network**: `mainnet`, `testnet`, or `devnet`
- **registry_address**: The existing registry contract address

The workflow will:
1. Build the codebase to generate genesis values (VK tree root, protocol contracts hash, etc.)
2. Deploy a new Rollup contract with the current code
3. Create a `RegisterNewRollupVersionPayload` that governance can execute to register the upgrade

## Network Configuration

All L1 contract parameters are defined in `spartan/environments/network-defaults.yml`. When you select a network:

- **Contract constants** (`AZTEC_EPOCH_DURATION`, `AZTEC_SLOT_DURATION`, thresholds, etc.) are loaded from the network's configuration
- **L1 chain** is inferred from `L1_CHAIN_ID` in the config (mainnet uses Ethereum mainnet, testnet/devnet use Sepolia)
- **Credentials** are fetched from GCP secrets based on the L1 chain (`sepolia-rpc-urls`, `sepolia-labs-rollup-private-key`, etc.)

This means the deployment automatically uses the correct parameters for each network without manual configuration.

## Follow-on tasks
- [TMNT-495](https://linear.app/aztec-labs/issue/TMNT-495/add-mainnet-labs-rollup-private-key-gcp-secret-for-rollup-upgrades): Add mainnet-labs-rollup-private-key GCP secret for rollup upgrades
